### PR TITLE
[WIP] ENH: Confounds for ConnectivityMeasure

### PR DESCRIPTION
**Pull request by KamalakerDadi**
This PR enhances the ConnectivityMeasure object by implementing confounds to regress out from the vectorized matrices. This is called as Connectome-level regression for example, to clean Mean FD.

### CODE REVIEW

File: nilearn/connectome/tests/test_connectivity_matrices.py [hunk 1]
```diff
@@ -595,3 +595,52 @@ def test_connectivity_measure_outputs():
...
+                                                n_confounds=5,
+                                                length=n_samples,
+                                                same_variance=False)
+        signals.append(signal)
+    correlation_measure = ConnectivityMeasure(kind='correlation',
+                                              vectorize=True)
+    # Clean confounds on 10 subjects with confounds filtered to 10 subjects in
+    # length
+    cleaned_vectors = correlation_measure.fit_transform(signals,
+                                                        confounds=confounds[0:10])
```
Comment (GaelVaroquaux) [inline nilearn/connectome/tests/test_connectivity_matrices.py]:
```suggestion
                                                        confounds=confounds[:n_subjects])
```

**Comment by GaelVaroquaux** [comment]
Looks great. I think that what we are missing is to use this in the example doing inter-subject comparison on resting-state.

For this, I would add the mean frame displacement per subject as a confound.

**Comment by kchawla-pi** [comment]
Hey @KamalakerDadi Pull from master.

**Comment by GaelVaroquaux** [comment]
Can you merge with master, so that CI runs without failing. Thanks!

**Comment by kchawla-pi** [comment]
Can't be merged without example.

**Comment by GaelVaroquaux** [comment]
We are getting trolled on twitter because this is not merged: https://twitter.com/fmrib_steve/status/1235204341310599169 :)

Let's merge it.

**Comment by kchawla-pi** [comment]
Does it need an example?

**Comment by GaelVaroquaux** [comment]
Well, it's better to have it in without an example than not to have it in.

I am resolving the merge conflicts.

**Comment by kchawla-pi** [comment]
You have always advocated against that slippery slope. We should establish a time table of by when we will add the example. Is there someone we can reach out to, to contribute one?

**Comment by GaelVaroquaux** [comment]
I resolved the conflicts. Will merge when

Wrt no example: it's a minor bit of code, and people are asking for it.

**Comment by emdupre** [comment]
I think @kchawla-pi 's point about targeting an example is a good one, even if we want to merge this PR without one.

What about opening an issue to note this for other (future) contributors ?

**Comment by bthirion** [comment]
Added issue  #2326

**Comment by kchawla-pi** [comment]
DO NOT Pull from master. There is some stuff that is going and until it has been attended to, do NOT pull from master.

**Comment by kchawla-pi** [comment]
Master is good now. All clear. You can go ahead and pull from it.

**Comment by thomasbazeille** [comment]
From what I get, the PR is good an issue is opened to add an example when possible. I think it can be merged in the meantime.

**Comment by jeromedockes** [comment]
> From what I get, the PR is good an issue is opened to add an example when possible. I think it can be merged in the meantime.

needs to be rebased but that should be easy as it's just a conflict in
what's new

**Comment by kchawla-pi** [comment]
Rebased? When did we switch to rebasing?

Whatsnew conflicts can be easily solved via the web editor right here.

**Comment by bthirion** [comment]
I think this one should be merged. Will do tomorrow unless I hear otherwise.

**Comment by bthirion** [comment]
Merfing, thx !

**Pull request merged**